### DISCUSSION TITLE
how do I access request headers in oak?

### DISCUSSION THREAD

**Discussion by ralyodio** [oakserver/oak#413] [Q&A]
I tried `request.headers.Authorization` but its not defined.

**Comment by ralyodio** [comment]
this gives error about value possibly being null:

```
  const auth = request.headers.get('Authorization');
  const matches = /Bearer\s*(.*)/.exec(auth);
  let jwt;

  if (matches.length > 1) {
    jwt = matches[1];
  }
```

error:

```
Check file:///home/www/grazily.com/grazily-api/index.ts
error: TS2345 [ERROR]: Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.
  const matches = /Bearer\s*(.*)/.exec(auth);
    
```

**Comment by ralyodio** [comment]
I"m trying to get the bearer token from the Authorization header

**Comment by aapoalas** [comment]
Check against null and only do the RegExp matching if the value truly is a string.

**Comment by kitsonk** [comment]
As mentioned, the error is correct, the value could be `null` and as @aapoalas says, you need to guard against it.  There are several ways to accomplish this, the easiest is:

```
const matches = /Bearer\s*(.*)/.exec(auth ?? "");
```
